sdk/spring/spring-cloud-azure-stream-binder-eventhubs/src/main/java/com/azure/spring/cloud/stream/binder/eventhubs/EventHubsMessageChannelBinder.java

```diff
@@ -281,21 +281,25 @@ InstrumentationManager getInstrumentationManager() {
     }
 
     /**
-     * Set the producer factory customizers.
+     * Add a producer factory customizer.
      *
-     * @param producerFactoryCustomizers The producer factory customizers.
+     * @param producerFactoryCustomizer The producer factory customizer to add.
      */
-    public void setProducerFactoryCustomizers(List<EventHubsProducerFactoryCustomizer> producerFactoryCustomizers) {
-        this.producerFactoryCustomizers = producerFactoryCustomizers;
+    public void addProducerFactoryCustomizer(EventHubsProducerFactoryCustomizer producerFactoryCustomizer) {
+        if (producerFactoryCustomizer != null) {
+            this.producerFactoryCustomizers.add(producerFactoryCustomizer);
```

Comment:
Do we need to consider thread-safe here?

Reply:
not for now